lib/mustache.ml

```diff
@@ -176,51 +176,86 @@ module Lookup = struct
 
 end
 
-let render_fmt ?(strict=true) (fmt : Format.formatter) (m : No_locs.t) (js : Json.t) =
+let render_fmt
+      ?(strict = true)
+      ?(partials = fun _ -> None)
+      (fmt : Format.formatter) (m : No_locs.t) (js : Json.t)
+  =
   let open No_locs in
   let add_context ctx js =
     match (ctx, js) with
     | `O ctx_o, `O js_o -> `O (ctx_o @ js_o)
     | _, _ -> ctx
   in
 
-  let rec render' m (js : Json.value) = match m with
+  let print_indent indent =
+    for i = 0 to indent - 1 do
+      Format.pp_print_char fmt ' '
+    done
+  in
+
+  let beginning_of_line = ref true in
+
+  let align indent =
+    if !beginning_of_line then (
+      print_indent indent;
+      beginning_of_line := false
+    )
+  in
+
+  let print_indented_string indent s =
+    for i = 0 to String.length s - 1 do
+      align indent;
+      Format.pp_print_char fmt s.[i];
+      if s.[i] = '\n' then (
+        beginning_of_line := true
+      )
+    done
+  in
```

Comment:
I changed `print_indented_string` to split the input on newlines and print line by line, instead of printing character by character.